test/DynamoCoreTests/Logging/DynamoAnalyticsClientTest.cs

```diff
@@ -0,0 +1,163 @@
+                selectionSet.AsEnumerable(),
+                Enumerable.Empty<Dynamo.Graph.Notes.NoteModel>(),
+                CurrentDynamoModel.CurrentWorkspace,
+                true,
+                new FunctionNamePromptEventArgs
+                {
+                    Category = "Testing",
+                    Description = "",
+                    Name = "__AnalyticsServiceTest__",
+                    Success = true
+                }) as CustomNodeWorkspaceModel;
+
+            //Act
+            //This will execute the custom workspace assigment and trigger the workspace assigment event
+            //The DynamoAnalyticsSession.Dispose() is executed automatically inside the Model and it will go to the crashing section.
+            CurrentDynamoModel.OpenCustomNodeWorkspace(customWorkspace.CustomNodeId);
+
+            //This will add a new custom node to the workspace
+            var ws = CurrentDynamoModel.CustomNodeManager.CreateCustomNode("someNode", "someCategory", "");
+            var csid = (ws as CustomNodeWorkspaceModel).CustomNodeId;
+            var customNode = CurrentDynamoModel.CustomNodeManager.CreateCustomNodeInstance(csid);
+
+            CurrentDynamoModel.AddNodeToCurrentWorkspace(customNode, false);
+
+            //This will get the value recorded in Registry about the crash/shoutdown
+            string foundRegValue = (string)Registry.GetValue(REG_KEY, SHUTDOWN_TYPE_NAME, DEFAULT_RETURN_VALUE);
+
+            DynamoModel.IsCrashing = false;
+
+            //Assert
+            Assert.AreEqual(CurrentDynamoModel.CurrentWorkspace.Nodes.Count(), InitialNodesCount + 1);
+            Assert.IsFalse(string.IsNullOrEmpty(foundRegValue));
+        }
+
+        /// <summary>
+        /// This test method execute the DynamoAnalyticsClient.GetUserID() to reach the end section
+        /// </summary>
+        [Test]
+        [Category("UnitTests")]
+        public void Test_DynamoAnalytics_GetUserID()
+        {
+            //Because surely the InstrumentationGUID entry in Registry already exists we need to delete it
+
+            //Arrange
+            // The name of the key must include a valid root.
+            const string userRoot = "HKEY_CURRENT_USER";
+            const string subkey = "Software\\DynamoUXG";
+            const string keyName = userRoot + "\\" + subkey;
+            string guid = string.Empty;
+
+            //First we try to get the value from the Windows Registry
+            var tryGetValue = Registry.GetValue(keyName, "InstrumentationGUID", null) as string;
+
+            if (tryGetValue != null)
+            {
+                //If the keyName already exists then we should remove it because the DynamoAnalyticsSession.GetUserID method will insert the value again
+                using (RegistryKey key = Registry.CurrentUser.OpenSubKey(subkey, true))
+                {
+                    key.DeleteValue("InstrumentationGUID");
```

Comment:
Does this require some kind of permission elevation?

Reply:
It's writing to the registry in the HKEY_CURRENT_USER key (current user logged), then we don't need extra permissions, also i ran the test cases in Visual Studio as a normal user and they ran successfully.